mikrotik/resource_bridge_port.go

```diff
@@ -0,0 +1,134 @@
+package mikrotik
+
+import (
+	"context"
+
+	"github.com/ddelnano/terraform-provider-mikrotik/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceBridgePort() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages ports in bridge associations.",
+
+		CreateContext: resourceBridgePortCreate,
+		ReadContext:   resourceBridgePortRead,
+		UpdateContext: resourceBridgePortUpdate,
+		DeleteContext: resourceBridgePortDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"bridge": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The bridge interface the respective interface is grouped in.",
+			},
+			"interface": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "*0",
+				Description: "Name of the interface.",
+			},
+			"pvid": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntBetween(1, 4094),
+				Description:  "Port VLAN ID (pvid) specifies which VLAN the untagged ingress traffic is assigned to. This property only has effect when vlan-filtering is set to yes.",
+			},
+			"comment": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Short description for this association.",
+			},
+		},
+	}
+}
+
+func resourceBridgePortCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*client.Mikrotik)
+	bridgePort, err := c.AddBridgePort(dataToBridgePort(d))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	recordBridgePortToData(bridgePort, d)
+	d.SetId(bridgePort.Id)
```

Comment:
`d.SetId()` is being called twice (within the `recordBridgePortToData` function), so we could omit it here.

Reply:
I was thinking about the place to put it, but apparently forgot to move it in another place so it left in both
Updated